### ISSUE TITLE
Bug: selection.getNodes() returns wrong data.

### ISSUE THREAD

**Issue by alirstm** [facebook/lexical#5261]
I'm trying to change headings back to paragraphs if the heading button is clicked again but selection.getNodes() returns `[TextNode]` instead of `[HeadingNode]`.

``` javascript
  const formatHeader = (tag: HeadingTagType) => {
    editor.update(() => {
      const selection = $getSelection()
      if ($isRangeSelection(selection)) {
        const nodes = selection.getNodes()

        // prints [TextNode] <----------------------- Here is the problem, where is HeadingNode.
        console.log(nodes)

        // prints [TextNode, HeadingNode]  <---------------- Ok, HeadingNode exists here but what's with TextNode? Also, it's not possible to use setBlocksType inside setTimeout.
        setTimeout(() => console.log(nodes), 0)

        if (selection.getNodes().some((n) => $isHeadingNode(n))) {
          $setBlocksType(selection, () => $createParagraphNode())
        } else {
          $setBlocksType(selection, () => $createHeadingNode(tag))
        }
      }
    })
  }
```

Lexical version: 0.12.4

## Steps To Reproduce

1. Go to playground.
2. Make a heading.
3. Try the "Clear Formatting" button.

It doesn't work even though it should. Here's playground's source code.

https://github.com/facebook/lexical/blob/303530a5f36ea9b54e6c3ba837e7ebb4665da8f2/packages/lexical-playground/src/plugins/ToolbarPlugin/index.tsx#L786

**Comment by GermanJablo** [comment]
The problem is that `selection.getNodes()` does not include a node if it does not have part of the selection `outside` the node, which seems to be a reasonable default behavior.

In `setBlocksType` I had to do something similar. 
https://github.com/facebook/lexical/blob/0f7c78b24dec932c9e1629623bedaf4485eacb08/packages/lexical-selection/src/range-selection.ts#L61-L68
I'm also doing that same thing in my code base. 

It occurs to me that `getNodes` could be given an optional parameter to include nodes that include the selection completely, up to an ancestor that meets a certain condition (because otherwise even `root` would be included). 

The lines I linked above and the ones you mention could be transformed into: `selection.getNodes(INTERNAL_$isBlock)`.
